src/Compilers/Test/Core/Diagnostics/DiagnosticExtensions.cs

```diff
@@ -273,6 +273,21 @@ public static TCompilation VerifyDiagnostics<TCompilation>(this TCompilation c,
             return c; // note this is a new compilation
         }
 
+        public static TCompilation VerifySuppressedAndFilteredDiagnostics<TCompilation>(
+            this TCompilation c,
+            DiagnosticAnalyzer[] analyzers,
+            AnalyzerOptions options = null,
+            Action<Exception, DiagnosticAnalyzer, Diagnostic> onAnalyzerException = null,
+            CancellationToken cancellationToken = default)
+            where TCompilation : Compilation
+        {
+            // Verify suppressed diagnostics are filtered when reportSuppressedDiagnostics is false.
+            c = c.GetCompilationWithAnalyzerDiagnostics(analyzers, options, onAnalyzerException, reportSuppressedDiagnostics: false, includeCompilerDiagnostics: true, cancellationToken, out var diagnostics);
+            diagnostics = diagnostics.WhereAsArray(d => d.IsSuppressed);
+            Assert.Empty(diagnostics);
```

Comment:
This seems to be a redundant check. Code below from [line 319-322](https://github.com/dotnet/roslyn/pull/53045/files#diff-a0b279d58233deddedac1fca458609ef81a83be228dee99e39476ec043b5ff87R319-R322) already does this validation.
```suggestion
```

Reply:
I mean the below existing code in `GetCompilationWithAnalyzerDiagnostics`:

```cs
            if (!reportSuppressedDiagnostics)
            {
                Assert.True(diagnostics.All(d => !d.IsSuppressed));
            }
```